_posts/2023-03-02-dafny-4-released.markdown

```diff
@@ -0,0 +1,98 @@
+---
+layout: post
+title:  "Dafny 4 is released"
+date:   2023-03-02 00:59:59 +0000
+---
+![Dafny Logo](/blog/assets/images/logo.jpg)
+
+Today, we are excited to announce the release of Dafny [4.0](https://github.com/dafny-lang/dafny/releases/tag/v4.0.0), a major release for the Dafny open-source programming language and automated reasoning tool!
+
+[**Dafny**](https://dafny.org) is a verification-aware programming language that has native support for capturing specifications and is equipped with a static program verifier. By blending sophisticated automated reasoning with familiar programming idioms and tools, Dafny empowers you to write provably correct code w.r.t. its specifications. In addition to a verification engine to check implementations against specifications, the Dafny ecosystem includes several compilers (C#, Java, JavaScript, Go, Python) that help you integrate Dafny into your existing workflows. The Dafny ecosystem also supports you with a Visual Studio Code [extension](https://marketplace.visualstudio.com/items?itemName=dafny-lang.ide-vscode), powered by an LSP implementation, and a code formatter. Dafny is used in academia for teaching and research, as well as in industry, such as by teams at Amazon (e.g. [AWS Encryption SDK](https://github.com/aws/aws-encryption-sdk-dafny)).
+
+**New Book!** The new book [_Program Proofs_](https://mitpress.mit.edu/9780262546232/program-proofs/) by Dafny creator, Rustan Leino, teaches you how to prove programs correct and uses Dafny throughout. It will be released by MIT Press in just a few days.
+
+For additional information on Dafny, please see:
+* [Manual Installation](https://github.com/dafny-lang/dafny/wiki/INSTALL) (or using the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=dafny-lang.ide-vscode))
+* [Dafny Reference Manual](https://dafny.org/latest/DafnyRef/DafnyRef)
+* [Dafny on GitHub](https://github.com/dafny-lang/dafny)
+
+## Dafny 4.0 Release Highlights
```

Comment:
The headings seem backwards to me. I'd make this one `#` and the following one `##`, and similar throughout.